File: 01-basic-challenges-1/09-count-vowels/count-vowels.js
function countVowels(string) {
  const lowerString = string.toLowerCase();
  const vowels = ["a", "e", "i", "o", "u", "y"];
  // const totalVowels = lowerString.split("").reduce((total, letter) => {
  //   if (vowels.includes(letter)) {
  //     total += 1;
  //   }
  //   return total;
  // }, 0);

  // OR
  // Filter all vowels out of the string made into array - and count items
  const vowelArray = lowerString.split("").filter((letter) => vowels.includes(letter));
  return vowelArray.length;
}

module.exports = countVowels;

File: 01-basic-challenges-1/09-count-vowels/count-vowels.rb
def countVowels(string)
  lower_string = string.downcase
  vowels = ["a", "e", "i", "o", "u", "y"]
  vowel_array = lower_string.chars.select { |letter| vowels.include?(letter) }
  vowel_array.length
end

puts countVowels("Hello World!")
